src/lib/download-manager.js

```diff
@@ -96,6 +97,53 @@ class DownloadManager {
 		}
 	}
 
+	async _validateCacheLimit({ url, maxCacheSize = settings.tachyonCacheLimit, currentFileName }) {
+		const fileHeaders = await fetch(url, { method: 'HEAD' });
+
+		const contentLength = parseInt(fileHeaders.headers.get('content-length') || '0', 10);
+		// get the size of the download directory
+		const downloadDirStats = await this.getDownloadedFilesStats(currentFileName);
+		const totalSize = (downloadDirStats.totalSize + contentLength) / (1024 * 1024 * 1024); // convert to GB
+		if (totalSize > maxCacheSize && downloadDirStats.fileStats.length > 0) {
+			// prompt to delete files
+			const question = {
+				type: 'confirm',
+				name: 'cleanCache',
+				message: 'Downloaded files exceed the cache limit. Do you want to clean up the cache?',
```

Comment:
This question should be reworded. The wording of "exceed the cache limit" is unnecessary.
"Do you want to delete previously downloaded versions to free up X GB?"

This means a method to calculate the list of files to be deleted and the size to be deleted needs to be extracted from _freeCacheSpace and called before the question.